src/cmdstan-guide/generate_quantities_config.qmd

```diff
@@ -1,21 +1,75 @@
-[QuickStart Guide section](generate_quantities_intro.qmd)
-and the Stan User's Guide section on
-[Stand-alone generated quantities and ongoing prediction](https://mc-stan.org/docs/stan-users-guide/posterior-prediction.html#stand-alone-generated-quantities-and-ongoing-prediction).
+Instead, you write a modified version of the original Stan program
+and add a generated quantities block or modify the existing one
+which specifies how to compute the new quantities of interest.
+Running the `generate_quantities` method on the new program
+together with sampler outputs (i.e., a set of draws)
+from the fitted model runs the generated quantities block
+of the new program using the the existing sample by plugging
+in the per-draw parameter estimates for the computations in
+the generated quantities block.
 
 This method requires sub-argument `fitted_params` which takes as its value
-an existing Stan CSV file that contains a sample from an equivalent model,
-i.e., a model with the same parameters, transformed parameters, and model blocks,
+an existing [Stan CSV](stan_csv_apdx.qmd) file that contains a parameter values
+from an equivalent model, i.e., a model with the same parameters and transformed parameters blocks,
 conditioned on the same data.
 
+The [generated quantities block](https://mc-stan.org/docs/reference-manual/blocks.html#generated-quantities)
+computes *quantities of interest* (QOIs) based on the data,
+transformed data, parameters, and transformed parameters.
+It can be used to:
+
+-  generate simulated data for model testing by forward sampling
+-  generate predictions for new data
+-  calculate posterior event probabilities, including multiple
+   comparisons, sign tests, etc.
+-  calculating posterior expectations
+-  transform parameters for reporting
+-  apply full Bayesian decision theory
+-  calculate log likelihoods, deviances, etc. for model comparison
+
+
+For an overview of the uses of this feature, see the Stan User's Guide section on
+[Stand-alone generated quantities and ongoing prediction](https://mc-stan.org/docs/stan-users-guide/posterior-prediction.html#stand-alone-generated-quantities-and-ongoing-prediction).
+
+
+## Example
+
+To illustrate how this works we use the `generate_quantities` method
+to do posterior predictive checks using the estimate of `theta` given
+the example bernoulli model and data, following the
+[posterior predictive simulation](https://mc-stan.org/docs/stan-users-guide/posterior-prediction.html#posterior-predictive-simulation-in-stan)
+procedure in the Stan User's Guide.
+
+We write a program `bernoulli_ppc.stan` which contains
+the following generated quantities block, with comments
+to explain the procedure:
+```stan
+generated quantities {
+  real<lower=0, upper=1> theta_rep;
+  array[N] int y_sim;
+  // use current estimate of theta to generate new sample
+  for (n in 1:N) {
+    y_sim[n] = bernoulli_rng(theta);
+  }
+  // estimate theta_rep from new sample
+  theta_rep = sum(y_sim) * 1.0 / N;
```

Comment:
Is there really no other way to do this than multiplying by 1.0?  

We can also move the declaration of `theta_rep` down to where it's defined now.

Reply:
Somehow we need to avoid int division, so this is probably the best current way.

Reply:
Right.  I proposed adding a div(x, y) function but it doesn't look like it ever got in.  We could also set theta_rep = sum(y_sim) and then set `theta_rep /= N` but that's way too clunky.  Maybe add a comment that the `1.0` is in there to convert an integer to a real.  Also, it's traditional to put the `1 / N` up front in an average.